### PR TITLE
bugfix: fix introspection and re-declaration of config

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | none
| License       | MIT
| Doc PR        | maybe not needed

So, I was trying GraphiQL, and wanted to develop an little test app with it see https://github.com/Simperfit/graphql-api/commit/b521052ac8903c6ccf11ae1c3e703ee458f3cf73 but I got into a little bug on the re-declaration of config, that fixed, i've seen another error with the declared type in the `SchemaBuilder`.

As I don't really followed the GraphQL development, I'm trying to figure out how to fix this.

### CODE REVIEW

File: features/graphql/introspection.feature [hunk 1]
```diff
@@ -168,6 +168,15 @@ Feature: GraphQL introspection support
...
+    """
+    And print last JSON response
```
Comment (Simperfit) [inline features/graphql/introspection.feature]: So this is failing with: 
```
│ {
      │     "errors": [
      │         {
      │             "debugMessage": "Unknown kind of type: StringIterableUnionInput",
      │             "message": "Internal server error",
      │             "category": "internal",
      │             "locations": [
      │                 {
      │                     "line": 62,
      │                     "column": 5
      │                 }
      │             ],
      │             "path": [
      │                 "__schema",
      │                 "types",
      │                 258,
      │                 "inputFields",
      │                 2,
      │                 "type",
      │                 "kind"
      │             ]
      │         },
      │         {
      │             "debugMessage": "Unknown kind of type: StringIterableUnionInput",
      │             "message": "Internal server error",
      │             "category": "internal",
      │             "locations": [
      │                 {
      │                     "line": 22,
      │                     "column": 5
      │                 }
      │             ],
      │             "path": [
      │                 "__schema",
      │                 "types",
      │                 259,
      │                 "kind"
      │             ]
      │         }
      │     ],
      │     "data": []
      │ }
```

In 2.2.5 this is not failing.

**Comment by Simperfit** [comment]
ping @alanpoulain for a review when you have time please.

**Comment by Simperfit** [comment]
Okay so this fixes the generation of the schemas, but it does not fix how GraphiQL expects the schema to be.
I'm getting alots of errors: 
`Error: "updateGerardInput.hellos field type must be Input Type but got: [StringIterableUnionInput]."`
`Error: "Introspection must provide object type for possibleTypes."`

Here is a valid reproducer : https://github.com/Simperfit/graphql-api/commit/bb1e22d1469ad93fafc30072cfab77ca259c1b0c

**Comment by alanpoulain** [comment]
I'm working on it. I've identified the issue and opened an issue in graphql-php. See https://github.com/webonyx/graphql-php/issues/272.

**Pull request closed without merging**